seismicpro/survey/headers.py

```diff
@@ -95,3 +96,103 @@ def callback(future, start_pos):
+                            "\n    positions for each trace")
+
+    if {*cdp_cols, *bin_cols} <= available_columns:
+        unique_inline_cdp = headers[[*bin_cols, *cdp_cols]].drop_duplicates()
+        unique_cdp = unique_inline_cdp[cdp_cols].drop_duplicates()
+        has_unique_inline_to_cdp = len(unique_inline_cdp) == len(unique_cdp)
+        if not has_unique_inline_to_cdp:
+            msg_list.append("Non-unique mapping of geographic (CDP_X, CDP_Y) to binary (INLINE_3D/"
+                            "\n    CROSSLINE_3D) coordinates")
+
+    if {*shot_cols, *rec_cols, *cdp_cols} <= available_columns:
+        raw_cdp = (headers[shot_cols].values + headers[rec_cols].values) / 2
+        has_consistent_geo_coords = np.allclose(raw_cdp, headers[cdp_cols].values, rtol=0, atol=cdp_atol)
+        if not has_consistent_geo_coords:
+            msg_list.append("Inconsistent range of some geographic coordinates (SourceX, SourceY, GroupX,"
+                            "\n    GroupY, CDP_X, CDP_Y)")
+
+    if {*shot_cols, "SourceSurfaceElevation"} <= available_columns:
+        ushot_elev_coords = headers[[*shot_cols, "SourceSurfaceElevation"]].drop_duplicates()
+        ushot_coords = ushot_elev_coords[shot_cols].drop_duplicates()
+        has_shot_uniq_elevs = len(ushot_elev_coords) == len(ushot_coords)
+        if not has_shot_uniq_elevs:
+            msg_list.append("Different surface elevation (SourceSurfaceElevation) for at least one shot")
+
+    if {*rec_cols, "ReceiverGroupElevation"} <= available_columns:
+        urec_elev_coords = headers[[*rec_cols, "ReceiverGroupElevation"]].drop_duplicates()
+        urec_coords = urec_elev_coords[rec_cols].drop_duplicates()
+        has_rec_uniq_elevs = len(urec_elev_coords) == len(urec_coords)
+        if not has_rec_uniq_elevs:
+            msg_list.append("Different surface elevation (ReceiverGroupElevation) for at least"
+                            "\n    one receiver")
+
+    if {*shot_cols, *rec_cols, "ReceiverGroupElevation", "SourceSurfaceElevation"} <= available_columns:
+        rec_elevations = urec_elev_coords.values[:, 2]
+        shot_interp = IDWInterpolator(ushot_elev_coords.values [:, :2], ushot_elev_coords.values[:, 2], neighbors=3)
+        rec_by_shot = shot_interp(urec_elev_coords.values[:, :2])
+        has_correct_elevations = np.allclose(rec_elevations, rec_by_shot, rtol=elev_rtol, atol=elev_atol)
+        if not has_correct_elevations:
+            msg_list.append("Inconsistent values in elevation-related headers (ReceiverGroupElevation,"
+                            "\n    SourceSurfaceElevation)")
+
+    if len(msg_list) > 0:
+        line = "\n\n" + "-"*80
+        msg = line +  "\n\nThe loaded Survey has the following problems with trace headers:"
+        msg += "".join([f"\n\n {i+1}. {msg}" for i, msg in zip(range(len(msg_list)), msg_list)]) + line
+        warnings.warn(msg, Warning)
```

Comment:
Is there a reason to use the base class `Warning` for your warnings instead of the default `UserWarning`?

```suggestion
        warnings.warn(msg)
```